client/src/app.js

```diff
@@ -191,15 +208,14 @@ export default function main({ DOM, HTTP, route, storage, scanner: scan$, search
 
   // Transaction analysis
   , txAnalysis$ = tx$.filter(Boolean)
-      .combineLatest(mempool$, feeEst$, (tx, mempool, feeEst) =>
-          ({ tx, mempool, feeEst, feerate: tx.fee ? tx.fee / tx.weight * 4 : null }))
-      .map(({ tx, feerate, mempool, feeEst }) => ({
+      .combineLatest(mempool$, feeEst$, calculateFeerates)
+      .map(({ tx, feerate, mempool, feeEst, effectiveFeerate}) => ({
         feerate
       , privacyAnalysis: getPrivacyAnalysis(tx)
       , segwitGains: calcSegwitFeeGains(tx)
-      , mempoolDepth: !tx.status.confirmed && feerate != null && mempool ? getMempoolDepth(mempool.fee_histogram, feerate) : null
-      , confEstimate: !tx.status.confirmed && feerate != null && feeEst ? getConfEstimate(feeEst, feerate) : null
-      , overpaying: !tx.status.confirmed && feerate != null && feeEst && feeEst[2] != null ? feerate/feeEst[2] : null
+      , mempoolDepth: !tx.status.confirmed && feerate != null && mempool ? getMempoolDepth(mempool.fee_histogram, effectiveFeerate) : null
+      , confEstimate: !tx.status.confirmed && feerate != null && feeEst ? getConfEstimate(feeEst, effectiveFeerate) : null
+      , overpaying: !tx.status.confirmed ? calculateOverpayment(effectiveFeerate, feeEst) : null
```

Comment:
Seems this should have the `feerate != null` test too

Reply:
thank you, applied.

Rebased to upstream, squashed into 1 commit.